### PR TITLE
chore: fix wasm file cache directory permission

### PR DESCRIPTION
**Pull request by zhaohuabing**
fixes: https://github.com/envoyproxy/gateway/pull/5819#issuecomment-2831091935

### CODE REVIEW

File: tools/docker/envoy-gateway/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
 FROM busybox@sha256:37f7b378a29ceb4c551b1b5582e27747b855bbfaa73fa11914fe0df028dc581f AS source
 # Create the data directory for eg
-RUN mkdir -p /var/lib/eg
+RUN mkdir -p /var/lib/eg && chmod -R 0777 /var/lib/eg
```
Comment (arkodg) [inline tools/docker/envoy-gateway/Dockerfile:3]: 0660 is prob safer, only user and group can read and write

File: tools/docker/envoy-gateway/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
 FROM gcr.io/distroless/base-nossl:nonroot@sha256:ecbab76d6a504ddf7c58a9d786e70f1f1731fa546b1ac0b20dab35c6fc2f3138
 ARG TARGETPLATFORM
 COPY --chown=65532:65532 $TARGETPLATFORM/envoy-gateway /usr/local/bin/
-COPY --from=source --chown=65532:65532 /var/lib /var/lib
```
[the remark this answers is not in the document]
Comment (zhaohuabing) [reply tools/docker/envoy-gateway/Dockerfile:10]: Ha, this shouldn't be required anymore.

[the remark this answers is not in the document]
Comment (zhaohuabing) [reply tools/docker/envoy-gateway/Dockerfile:10]: Ha, it should because the user can be changed in the container SecurityContex.

File: test/e2e/testdata/custom-container-security-contex-userid.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  name: custom-container-security-contex-userid-envoyproxy
+  namespace: gateway-conformance-infra
+spec:
+  provider:
+    type: Kubernetes
+    kubernetes:
+      envoyDeployment:
+        container:
+          securityContext:
+            runAsUser: 65534
```
Comment (arkodg) [inline test/e2e/testdata/custom-container-security-contex-userid.yaml:13]: the related securityContext is tied to EG Deployment, not Envoy

Comment (zhaohuabing) [reply test/e2e/testdata/custom-container-security-contex-userid.yaml:13]: This is envoy deployment:

```yaml
apiVersion: [gateway.envoyproxy.io/v1alpha1](http://gateway.envoyproxy.io/v1alpha1)
kind: EnvoyProxy
metadata:
  name: custom-container-security-contex-userid-envoyproxy
  namespace: gateway-conformance-infra
spec:
  provider:
    type: Kubernetes
    kubernetes:
      envoyDeployment:     # envoy deployment, for both the envoy and shutdown manager Security Context
        container:
          securityContext:
            runAsUser: 65534
            runAsGroup: 65534
```

File: test/e2e/tests/httproute_with_custom_security_context_userid.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+
+			http.MakeRequestAndExpectEventuallyConsistentResponse(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr, expectedResponse)
+
+			// we don't set the envoy-gateway deployment security context user id back to default because this will
+			// cause the envoy proxies failed to be deleted after the Gateway resources in the base are deleted.
+			// This is acceptable because this won't affect the later tests in the same suite.
```
Comment (zhaohuabing) [inline test/e2e/tests/httproute_with_custom_security_context_userid.go]: ~~This happens because the Envoy Gateway may lose the deletion message of Gateways while rollout restart, and the left Envoy proxies cause failure of the upgrade test.~~

~~This is an edge case found in this test  - should EG proactively delete the Envoy proxies that are orphaned after Gateways are removed?~~

Manually delete the Envoy proxies after the test.

Related issue: https://github.com/envoyproxy/gateway/issues/3051

https://github.com/envoyproxy/gateway/actions/runs/15431221218/job/43429760632?pr=6173

```
➜  gateway git:(fix-cache-permission) ✗ k -n envoy-gateway-system get pod -w
NAME                                                              READY   STATUS    RESTARTS   AGE
envoy-gateway-79df56475-946vz                                     1/1     Running   0          29s
envoy-gateway-conformance-infra-all-namespaces-302def45-5dj9bxt   1/2     Running   0          42s
envoy-gateway-conformance-infra-same-namespace-9baff503-69cchq4   1/2     Running   0          42s
```

**Review by arkodg — APPROVED** [review]
LGTM, thanks for adding the e2e

**Review by cnvergence — APPROVED** [review]

**Review by zirain — APPROVED** [review]
